.github/workflows/extract_versions.yml

```diff
@@ -0,0 +1,37 @@
+name: Extract versions
+
+on:
+  workflow_call:
+    inputs:
+      ignore_prefix:
+        required: false
+        default: ''
+        type: string
+    outputs:
+      major_version: 
+        value: ${{ jobs.extract_versions.outputs.major_version }}
+      minor_version: 
+        value: ${{ jobs.extract_versions.outputs.minor_version }}
+      full_version: 
+        value: ${{ jobs.extract_versions.outputs.full_version }}
+
+jobs:
+  extract_versions:
+    runs-on: ubuntu-latest
+    outputs:
+      minor_version: ${{ steps.vars.outputs.minor_version }}
+      major_version: ${{ steps.vars.outputs.major_version }}
+      full_version: ${{ steps.vars.outputs.full_version }}
+    steps:
+      - name: extract major and minor version substrings
+        id: vars
+        run: |
+          MAJOR_VERSION="$(echo "${GITHUB_REF_NAME##${{ inputs.ignore_prefix }}}" | cut -d '.' -f 1)"
```

Comment:
does `##` mean removing whatever specified on the right from the left side?

Reply:
In this case the `##` removes the inputs.ignore_prefix from the `GITHUB_REF_NAME`. So if the tag is search-bar-v1.12.0, the prefix would be ignored and you would get v1 as the major version. If the ignore_prefix is empty, the GITHUB_REF_NAME will be used (no effect)